src/ape/api/query.py

```diff
@@ -61,32 +81,47 @@ class AccountQuery(_BaseAccountQuery):
     of transactions made by ``account`` between ``start_nonce`` and ``stop_nonce``.
     """
 
-    type: Literal["accounts"] = "accounts"
     account: AddressType
 
+    @classmethod
+    def all_fields(cls) -> List[str]:
+        return list(TransactionAPI.__fields__)
+
 
 class ContractEventQuery(_BaseBlockQuery):
     """
     A ``QueryType`` that collects members from ``event`` over a range of
     logs emitted by ``contract`` between ``start_block`` and ``stop_block``.
     """
 
-    type: Literal["contract_events"] = "contract_events"
     contract: AddressType
     event: EventABI
 
+    @classmethod
+    def all_fields(cls) -> List[str]:
+        # TODO: Figure out how to get the event ABI as a class property
+        #   for the validator
+        return [
+            i.name for i in cls.event.inputs if i.name is not None
+        ]  # if i.name is not None just for mypy
+
 
 class ContractMethodQuery(_BaseBlockQuery):
     """
     A ``QueryType`` that collects return values from calling ``method`` in ``contract``
     over a range of blocks between ``start_block`` and ``stop_block``.
     """
 
-    type: Literal["contract_calls"] = "contract_calls"
     contract: AddressType
     method: MethodABI
     method_args: Dict[str, Any]
 
+    @classmethod
+    def all_fields(cls) -> List[str]:
+        # TODO: Figure out how to get the event ABI as a class property
```

Comment:
Is this TODO only supposed to be on the class above (ContractEventQuery) and not here?
Or should this say `# TODO: Figure out how to get the method ABI as a class property`

Reply:
```suggestion
        # TODO: Figure out how to get the method ABI as a class property
```